dan_lombardy/lib/updateDoc.js

```diff
@@ -0,0 +1,9 @@
+var Message = require(__dirname + '/../models/message');
+var errorHandler = require(__dirname + '/errorHandler');
+
+module.exports = function(doc, change, res){
+  Message.findOneAndUpdate(doc, change, {new:true}, function(err, doc){
+    errorHandler(err, res);
```

Comment:
Although checking for the error in the error handler looks efficient codewise, calling functions is relatively expensive. Running is more efficient if you check whether you need to use a function before executing it.